cli/flox-rust-sdk/src/environment.rs

```diff
@@ -17,7 +17,7 @@ pub static NIX_BIN: &str = env!("NIX_BIN");
 /// For flox specifically, set Nix-provided defaults for certain
 /// environment variables that we know to be required on the various
 /// operating systems.
-pub fn default_nix_subprocess_env() -> HashMap<String, String> {
+pub fn default_nix_subprocess_env(set_all: bool) -> HashMap<String, String> {
```

Comment:
nonblocking: add to the comment above describing what `set_all` does

Reply:
Added comment - @mkenigs let me know if you'd want more by way of a comment here.